hello-thirdparty/app/build.gradle

```diff
@@ -18,42 +29,34 @@ model {
         }
     }
 
-    compileOptions.with {
-        sourceCompatibility=JavaVersion.VERSION_1_7
-        targetCompatibility=JavaVersion.VERSION_1_7
-    }
-
     android.ndk {
         moduleName = "native-activity"
-        cppFlags.addAll(["-I${file("src/main/jni")}".toString(),
-	                 "-I${gpg_sdk_path}/include".toString()])
+        platformVersion = 9 //same as minSdkVersion.apiLevel for better compatibility
         stl    = "c++_static"
-        ldLibs.addAll(["log", "android", "EGL", "GLESv1_CM", "z"])
-        //
-        //  NOTE: since libgpg is also linked to the stl:
-        //  libc++ explicitly follows it in the list of libraries
-        //  to prevent unresolved stl symbols.
-        //
-        ldLibs.addAll(["gpg", "c++_static"])
+        ldLibs.addAll(["atomic", "log", "android", "EGL", "GLESv1_CM", "z"])
```

Comment:
gpg depends on atomic, log and z (according to gpg lib's original sample Android.mk).
and the rest of the code depends on log, android, EGL and GLESv1_CM.
Since some dependencies are common (log), having a separate call to addAll with all the deps would lead to add these libs twice.
I've added a comment to explicit the dependency. We can have a separate call to add atomic and z too if you prefer.